user_modules/api/APITestDataGenerator.lua

```diff
@@ -0,0 +1,308 @@
+----------------------------------------------------------------------------------------------------
+-- API Test Data Generator module
+----------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]-------------------------------------------------------------------
+local ah = require("user_modules/api/APIHelper")
+
+--[[ Module ]]--------------------------------------------------------------------------------------
+local m = {}
+
+--[[ Constants ]]-----------------------------------------------------------------------------------
+m.valueType = {
+  VALID_RANDOM = 1,
+  LOWER_IN_BOUND = 2,
+  UPPER_IN_BOUND = 3,
+  LOWER_OUT_OF_BOUND = 4,
+  UPPER_OUT_OF_BOUND = 5,
+  INVALID_TYPE = 6
+}
+
+--[[ Value generators ]]----------------------------------------------------------------------------
+math.randomseed(tonumber(tostring(os.time()):reverse():sub(1,6)))
+
+--[[ @getStringValue: Return value for 'string' data type
+--! @parameters:
+--! pTypeData: table with data type restrictions
+--! @return: value for the parameter
+--]]
+local function getStringValue(pTypeData)
+  local length
+  if pTypeData.valueType == m.valueType.LOWER_IN_BOUND then
+    length = pTypeData.minlength
```

Comment:
Is this minlength taken from API for the parameter in question?

Reply:
Correct.